### PR TITLE
Fix bug where animations had more keyframes than the resource

### PR DESCRIPTION
**Pull request by mariofv**
What's new:
- Changes in between frames interpolation. 
- Changes in animation importation keyframes number

### CODE REVIEW

File: Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp [hunk 1]
```diff
@@ -85,6 +83,7 @@ void AnimationImporter::GetCleanAnimation(const aiNode* root_node, const aiAnima
 		own_format_animation.keyframes.push_back({ static_cast<float>(keyframe.first), keyframe.second });
 	}
 
+	//assert(own_format_animation.keyframes.size() == animation_duration);
```
Comment (yalania) [inline Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp]: if it is not require anymore better remove it

File: Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp [hunk 2]
```diff
@@ -58,24 +59,21 @@ void AnimationImporter::GetCleanAnimation(const aiNode* root_node, const aiAnima
...
+			GetChannelTransform(pre_transform, channel, animation_duration, channel_transform);
 		}
 
-		for (size_t i = 0; i <= animation_duration; ++i)
+		for (size_t i = 0; i < animation_duration; ++i)
 		{
-			float3 euler_rotation = channel_transform[i].ToEulerXYX();
-			Quat rotation = Quat::FromEulerXYX(euler_rotation.x, euler_rotation.y, euler_rotation.z);
+			float3 euler_rotation = channel_transform[i].ToEulerXYZ();
+			Quat rotation = Quat::FromEulerXYZ(euler_rotation.x, euler_rotation.y, euler_rotation.z);
```
Comment (yalania) [inline Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp:72]: I don't understand how this was working xD

**Review by yalania — APPROVED** [review]
Thanks a lot for taking care of this :D

File: Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp (hunk 2 above)
Comment (mariofv) [reply Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp:72]: : D

File: Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp (hunk 1 above)
Comment (mariofv) [reply Engine/ResourceManagement/Importer/ModelImporters/AnimationImporter.cpp]: True, changing it right now!

**Pull request merged**